src/domain/core/src/services/push_service.rs

```diff
@@ -40,7 +40,7 @@ pub trait PushService: Send + Sync {
 #[derive(Debug, Clone)]
 pub struct PushRequest {
     pub local_ref: Option<DatasetRef>,
-    pub remote_ref: Option<DatasetRefRemote>,
+    pub remote_ref: Option<TransferDatasetRef>,
```

Comment:
Hmm, it's now strange to see this type here. 

If we want `remote_ref` to represent a *resolved* push destination, i.e. **where exactly** the push is going to, then perhaps it should be a struct containing a Url, and optional repo, account, and dataset name?

And perhaps this what `RemoteAliasResolver::resolve_remote_alias` should be returning?

What do you think?